ir/src/test/scala/org/scalajs/core/ir/PrintersTest.scala

```diff
@@ -0,0 +1,999 @@
+package org.scalajs.core.ir
+
+import scala.language.implicitConversions
+
+import org.junit.Test
+import org.junit.Assert._
+
+import Definitions._
+import Printers._
+import Trees._
+import Types._
+
+import OptimizerHints.{empty => NoOptHints}
+
+class PrintersTest {
+  private implicit val dummyPos = Position.NoPosition
+
+  private def assertPrintEquals(expected: String, tree: Tree): Unit =
+    assertPrintEqualsImpl(expected, _.print(tree))
+
+  private def assertPrintEquals(expected: String, tpe: Type): Unit =
+    assertPrintEqualsImpl(expected, _.print(tpe))
+
+  private def assertPrintEqualsImpl(expected: String,
+      print: IRTreePrinter => Unit): Unit = {
+    val sw = new java.io.StringWriter
+    val printer = new IRTreePrinter(sw)
+    print(printer)
+    assertEquals(expected.stripMargin.trim, sw.toString())
+  }
+
+  private implicit def string2ident(name: String): Ident = Ident(name)
+  private implicit def string2classType(cls: String): ClassType = ClassType(cls)
+
+  private def b(value: Boolean): BooleanLiteral = BooleanLiteral(value)
+  private def i(value: Int): IntLiteral = IntLiteral(value)
+  private def l(value: Long): LongLiteral = LongLiteral(value)
+  private def f(value: Float): FloatLiteral = FloatLiteral(value)
+  private def d(value: Double): DoubleLiteral = DoubleLiteral(value)
+
+  private def ref(ident: Ident, tpe: Type): VarRef = VarRef(ident)(tpe)
+
+  @Test def printType(): Unit = {
+    assertPrintEquals("any", AnyType)
+    assertPrintEquals("nothing", NothingType)
+    assertPrintEquals("void", UndefType)
+    assertPrintEquals("boolean", BooleanType)
+    assertPrintEquals("int", IntType)
+    assertPrintEquals("long", LongType)
+    assertPrintEquals("float", FloatType)
+    assertPrintEquals("double", DoubleType)
+    assertPrintEquals("string", StringType)
+    assertPrintEquals("null", NullType)
+    assertPrintEquals("<notype>", NoType)
+
+    assertPrintEquals("O", ClassType(ObjectClass))
+
+    assertPrintEquals("O[]", ArrayType(ObjectClass, 1))
+    assertPrintEquals("I[][]", ArrayType("I", 2))
+
+    assertPrintEquals("(x: int, var y: any)",
+        RecordType(List(
+            RecordType.Field("x", None, IntType, mutable = false),
+            RecordType.Field("y", None, AnyType, mutable = true))))
+  }
+
+  @Test def printEmpty(): Unit = {
+    assertPrintEquals("<empty>", EmptyTree)
+  }
+
+  @Test def printVarDef(): Unit = {
+    assertPrintEquals("val x: int = 5",
+        VarDef("x", IntType, mutable = false, i(5)))
+    assertPrintEquals("var x: int = 5",
+        VarDef("x", IntType, mutable = true, i(5)))
```

Comment:
Please test the case of an empty rhs (we shouldn't print the equality sign in that case).